share/functions/__fish_config_interactive.fish

```diff
@@ -232,6 +232,21 @@ function __fish_config_interactive -d "Initializations that should be performed
     # Load key bindings
     __fish_reload_key_bindings
 
+    if not set -q FISH_UNIT_TESTS_RUNNING
+        # Enable bracketed paste before every prompt (see __fish_shared_bindings for the bindings).
+        # Disable it for unit tests so we don't have to add the sequences to bind.expect
+        function __fish_enable_bracketed_paste --on-event fish_prompt
+            printf "\e[?2004h"
+        end
+
+        # Disable BP before every command because that might not support it.
+        function __fish_disable_bracketed_paste --on-event fish_preexec
+            printf "\e[?2004l"
+        end
+
+        __fish_enable_bracketed_paste
```

Comment:
Of course it's needed - we need to tell the terminal we support bracketed paste, and since this is in __f_c_i, the first fish_prompt has already fired. Will add a comment to that effect.